lib/matplotlib/sphinxext/plot_directive.py

```diff
@@ -633,8 +639,9 @@ def run(arguments, content, options, state_machine, state, lineno):
             source_file_name = os.path.join(setup.confdir, config.plot_basedir,
                                             directives.uri(arguments[0]))
 
-        # If there is content, it will be passed as a caption.
-        caption = '\n'.join(content)
+        # If there is content, it will be passed as a caption, if the caption
+        # option is not present.
+        caption = options.get("caption", '\n'.join(content))
```

Comment:
I can see an argument here for raising an exception if there is a collision.  It can be very tricky to track down what exactly sphinx is doing under the hood and failing hard when we have to guess should make it easier to identify where we are guessing differently than the user wished.

Reply:
Is it correct that the collision can only happen in a case similar to the example
~~~
.. plot:: range4.py
   :caption: This is the actual caption for plot 18.

   This is the caption for plot 18.
~~~
where the `:caption:` and the text-as-caption are both part of the plot directive and thus close to each other?

If that's the case, I'm 👍 on raising because this is an ambiguous API use. It should be and can be easily clarified on the user side.

Reply:
Thanks for reviewing! I see the argument for raising. I'll put that in later today.